lib/ransack/visitor.rb

```diff
@@ -44,7 +44,18 @@ def visit_or(object)
     end
 
     def visit_Ransack_Nodes_Sort(object)
-      object.attr.send(object.dir) if object.valid?
```

Comment:
This might be a good place for a guard conditional with early return like `return unless object.valid?`.